calicoctl/commands/status.go

```diff
@@ -0,0 +1,206 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/docker/engine-api/client"
+	"github.com/docker/engine-api/types"
+	"github.com/docopt/docopt-go"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+var ipv4 = true
+
+// Status prings status of the node and returns error (if any)
+func Status(args []string) error {
+	doc := `Usage:
+calicoctl node status
+
+Description:
+  Display the status of the calico node`
+
+	_, _ = docopt.Parse(doc, args, true, "calicoctl", false, false)
+
+	ctx := context.Background()
+
+	defaultHeaders := map[string]string{"User-Agent": "engine-api-cli-1.0"}
+	cli, err := client.NewClient("unix:///var/run/docker.sock", "v1.22", nil, defaultHeaders)
+	if err != nil {
+		panic(err)
+	}
+
+	options := types.ContainerListOptions{All: true}
+	containers, err := cli.ContainerList(ctx, options)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, c := range containers {
+		if c.Names[0] == "/calico-node" && c.State == "running" {
+
+			fmt.Printf("calico-node container is running. Status: %s\n", c.Status)
+
+			// config := types.ExecConfig{
+			// 	User:         "root",
+			// 	AttachStdin:  true,
+			// 	AttachStderr: true,
+			// 	AttachStdout: true,
+			// 	//Detach:       true,
+			// 	Cmd: []string{"sh", "-c", "cat libraries.txt"},
+			// }
+
+			// exec, err := cli.ContainerExecCreate(context.Background(), c.ID, config)
+			// if err != nil {
+			// 	fmt.Printf("failed to create exec: %v", err)
+			// 	break
+			// }
+
+			// reply, err := cli.ContainerExecAttach(context.Background(), exec.ID, config)
+			// if err != nil {
+			// 	fmt.Printf("failed to attach to the container: %v", err)
+			// 	break
+			// }
+			// defer reply.Close()
+
+			// b := make([]byte, 255)
+
+			// _, err = reply.Reader.Read(b)
+			// if err != nil {
+			// 	fmt.Printf("failed to exececute command inside the container: %v", err)
+			// 	break
+			// }
+
+			//
+			// re, err := regexp.Compile(`calico\s(.*)\n`)
+			// if err != nil {
+			// 	fmt.Printf("failed to compile regex: %v", err)
+			// 	break
+			// }
+
+			// parts := strings.Fields(re.FindString(string(b)))
+			// fmt.Println(parts)
+
+			// if len(parts) == 2 {
+			// 	felixVersion := parts[1][1 : len(parts[1])-1]
+
+			// 	fmt.Printf("Running felix version %v\n", felixVersion)
+			// }
+
+			//
+
+			//if calico-node container is running, then connect to the bird socket and get the data
```

Comment:
yeah, I think it would make sense to refactor the code/encapsulate when we do the etcd part. i.e. in GA